lib/router.js

```diff
@@ -40,41 +40,40 @@ exports.setupModelRoutes = function () {
 
 	// model routes
 	router.get('/models/list', routes.modelList);
-	router.get('/model/:model/list', middleware.modelIndex.get, routes.modelIndex);
-    router.post('/model/:model/list', middleware.modelIndex.post, routes.modelIndex);
-	router.get('/model/:model/new', middleware.modelCreate(), routes.modelCreate);
-	router.post('/model/:model/create', middleware.modelSave(), routes.modelSave);
-    router.get('/model/:model/export', middleware.modelExport.get, routes.modelExport.get);
-    router.post('/model/:model/export', middleware.modelExport.post);
+	router.get('/model/:model/list', middleware.model, middleware.permissions('index', 'model'), middleware.modelIndex.get, routes.modelIndex);
+    router.post('/model/:model/list', middleware.model, middleware.permissions('index', 'model'), middleware.modelIndex.post, routes.modelIndex);
```

Comment:
I don't know why I didn't think of this before, but we could replace all of the `middleware.model` and `middleware.config` calls with Express's `app.param` functionality (http://expressjs.com/api.html#app.param). It would reduce the verbosity of our code.